SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -0,0 +1,505 @@
+#include "PokemonHome_BoxSorting.h"
+#include "CommonFramework/ImageTools/ImageBoxes.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonHome{
+    using namespace Pokemon;
+
+namespace{
+}
+
+BoxSorting_Descriptor::BoxSorting_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonHome:BoxSorter",
+        STRING_POKEMON + " Home", "Box Sorter",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonHome/BoxSorter.md",
+        "Order boxes of " + STRING_POKEMON + ".",
+        FeedbackType::REQUIRED, false,
+        PABotBaseLevel::PABOTBASE_12KB
+    )
+{}
+
+BoxSorting::BoxSorting()
+        : BOX_NUMBER(
+            "<b>Number of boxes to order:</b>",
+            1
+        ),
+        VIDEO_DELAY(
+            "<b>Delay of your capture card (you probably have to increase this):</b>",
+            20
+        ),
+        GAME_DELAY(
+            "<b>Delay of your Pokemon Home app (default value should be fine):</b>",
+            10
+        )
+    {
+        PA_ADD_OPTION(BOX_NUMBER);
+        PA_ADD_OPTION(VIDEO_DELAY);
+        PA_ADD_OPTION(GAME_DELAY);
+    }
+
+bool go_to_first_slot(SingleSwitchProgramEnvironment& env, BotBaseContext& context, uint16_t video_delay){
+
+    PokemonAutomation::ImageFloatBox Cursor_check(0.07, 0.15, 0.01, 0.01); //cursor position of the first slot of the box
+    std::shared_ptr<const ImageRGB32> screen = env.console.video().snapshot();
+    VideoOverlaySet BoxRender(env.console);
+
+    BoxRender.add(COLOR_BLUE, Cursor_check);
+    if(image_stats(extract_box_reference(*screen, Cursor_check)).average.r != 255){
+
+        bool cursor_found = false;
+
+        for (uint8_t lines = 0; lines < 7; lines++) {
+            for (uint8_t colomn = 0; colomn < 5; colomn++) {
+                pbf_press_dpad(context, DPAD_LEFT, 10, video_delay);
+                context.wait_for_all_requests();
+                screen = env.console.video().snapshot();
+
+                if(image_stats(extract_box_reference(*screen, Cursor_check)).average.r == 255){
+                    cursor_found = true;
+                    break;
+                }
+            }
+            if(!cursor_found){
+                pbf_press_dpad(context, DPAD_UP, 10, video_delay);
+                context.wait_for_all_requests();
+                screen = env.console.video().snapshot();
+
+                if(image_stats(extract_box_reference(*screen, Cursor_check)).average.r == 255){
+                    cursor_found = true;
+                    break;
+                }
+            }else{
+                break;
+            }
+        }
+        if(!cursor_found){
+            return false;
+        }
+    }
+    BoxRender.clear();
+    return true;
+}
+
+struct Cursor{
+  uint16_t box;
+  uint16_t colomn;
+  uint16_t line;
+};
+
+
+
+void swap(std::vector<uint16_t>* arr, int pos1, int pos2){
+    int temp;
+    temp = (*arr)[pos1];
+    (*arr)[pos1] = (*arr)[pos2];
+    (*arr)[pos2] = temp;
+}
```

Comment:
you can remove this and use std::swap
https://en.cppreference.com/w/cpp/algorithm/swap
you have an example here https://github.com/pifopi/Arduino-Source/blob/65f55c1b19c0f12d4f5eee546392d5c7e8e16af1/SerialPrograms/Source/PokemonSwSh/Programs/General/PokemonSwSh_BoxReorderNationalDex.cpp#L226